Implement top2vec topic modeling based on multilingual LaBSE embeddings

NorDar opened the pull request:
This PR will add functionality to fit a top2vec model pipeline, save the model, predict topics and save the dataframe with the new topics to disk.

Additionally some documentation of the hyper parameter optimization was added. This needs to be reworked for the new, 3 year dataset.

NorDar committed:
remove articles below 400 characters, these are artifacts, videos, pic dumps etc.

NorDar committed:
topic modelling by multilingual embedding with top2vec including hyper paramter optimization

NorDar committed:
Merge branch 'main' into labse-embeddings

Apply lda and lsa changes.

NorDar committed:
transform all queries at the same time. save time, don't spam the output with log entries.

NorDar committed:
implement top2vec topic modeling with multilingual LaBSE embeddings.

Danielgergely commented on main.py:
```diff
@@ -83,6 +89,36 @@
+            or force_recreate
+            or _previous_step_recreate
+    ):
+        top2vec_model = TopicModelingService.fit_top2vec_model(data_column=sentiment_df["content"])
```
Do we maybe want to use the same syntax as in the other services? 

Service.method(df:pd.DataFrame, column_to_process:str="content")?

NorDar replied:
I think for a production ready application standardization would make sense. But for now I would rather not retest. 
Seems like everything is working. 

Also I think in general I would do it the other way around. Pass only the columns the method needs. Experience shows that it is already only one column. Then return only one column.

NorDar committed:
hyperparameter optimization results

NorDar committed:
hyper parameter tuning and analysis of tuned models topic correlation

NorDar committed:
add correlation metrics to analysis csv

NorDar committed:
analysis of topic content and wordclouds
analyzing top 5 positive and top 5 negativ correlated topics between normalized, mean of de and fr
